api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java

```diff
@@ -0,0 +1,129 @@
+package org.pmiops.workbench.billing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.pmiops.workbench.db.model.BillingProjectBufferEntry.BillingProjectBufferStatus.ASSIGNED;
+import static org.pmiops.workbench.db.model.BillingProjectBufferEntry.BillingProjectBufferStatus.CREATING;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import javax.inject.Provider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockitoAnnotations;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchConfig.FireCloudConfig;
+import org.pmiops.workbench.db.dao.BillingProjectBufferEntryDao;
+import org.pmiops.workbench.db.model.BillingProjectBufferEntry;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.test.FakeClock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Import(LiquibaseAutoConfiguration.class)
+public class BillingProjectBufferServiceTest {
+
+  private static final Instant NOW = Instant.now();
+  private static final FakeClock CLOCK = new FakeClock(NOW, ZoneId.systemDefault());
+
+  private static WorkbenchConfig workbenchConfig;
+
+  @TestConfiguration
+  @Import({})
+  @MockBean({
+      FireCloudService.class
+  })
+  static class Configuration {
+    @Bean
+    @Scope("prototype")
+    public WorkbenchConfig workbenchConfig() {
+      return workbenchConfig;
+    }
+  }
+
+
+  @Autowired
+  private BillingProjectBufferEntryDao billingProjectBufferEntryDao;
+  @Autowired
+  private FireCloudService fireCloudService;
+  @Autowired
+  private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  private BillingProjectBufferService billingProjectBufferService;
+
+  private final long BUFFER_CAPACITY = 5;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    billingProjectBufferService = new BillingProjectBufferService(
```

Comment:
It seems like you should be able to just export a `@Bean Clock` from your test config, then just `@Autowire BillingProjectBufferService`, and avoid the manual construction.